substra/sdk/backends/local/db.py

```diff
@@ -36,11 +36,8 @@ def add(self, asset):
         else:
             key = asset.key
 
-        if key in self._data[type_]:
-            raise exceptions.AlreadyExists(key, 409)
-        else:
-            self._data[type_][key] = asset
-            logger.info(f"{type_} with key '{key}' has been created.")
+        self._data[type_][key] = asset
+        logger.info(f"{type_} with key '{key}' has been created.")
 
         return asset
 
```

Comment:
this about the "get" function below, the "log" argument is now useless (it was to test the existence before adding without logging an error "already exists")

Reply:
The `log argument is still used in the KeyError case

```
    def get(self, type_, key: str, log: bool = True):
        """Return asset."""
        try:
            return self._data[type_][key]
        except KeyError:
            if log:
                logger.error(f"{type_} with key '{key}' not found.")
            raise exceptions.NotFound(f"Wrong pk {key}", 404)
```

do you want me to remove that `logger.error`?

Reply:
On the contrary, now that there is no check in the local backend `add` function, we can always do `logger.error` when the asset  does not exist. The `log` parameter to hide the error message is useless.